cudamapper/src/index_gpu.cuh

```diff
@@ -563,6 +577,57 @@ IndexGPU<SketchElementImpl>::IndexGPU(std::shared_ptr<DeviceAllocator> allocator
                    filtering_parameter);
 }
 
+template <typename SketchElementImpl>
+IndexGPU<SketchElementImpl>::IndexGPU(std::shared_ptr<DeviceAllocator> allocator,
+                                      const IndexCache& host_cache)
+    : first_read_id_(host_cache.first_read_id())
+    , kmer_size_(host_cache.kmer_size())
+    , window_size_(host_cache.window_size())
+    , allocator_(allocator)
+    , representations_d_(allocator)
+    , read_ids_d_(allocator)
+    , positions_in_reads_d_(allocator)
+    , directions_of_reads_d_(allocator)
+    , unique_representations_d_(allocator)
+    , first_occurrence_of_representations_d_(allocator)
+{
+    number_of_reads_                     = host_cache.number_of_reads();                     //H2D
+    number_of_basepairs_in_longest_read_ = host_cache.number_of_basepairs_in_longest_read(); //H2D
+
+    //H2D- representations_d_ = host_cache.representations();
+    representations_d_.resize(host_cache.representations().size());
+    representations_d_.shrink_to_fit();
```

Comment:
FYI: Currently `device_buffer::resize()` allocates exactly the needed amount of memory: https://github.com/clara-genomics/ClaraGenomicsAnalysis/blob/b5ea4f4d9d11ccf2bd8e8b4a571168ba6a996547/common/utils/include/claragenomics/utils/buffer.hpp#L148
`device_buffer::shrink_to_fit()` performs the check and only shrinks if need, so in this case it does not have any effect: https://github.com/clara-genomics/ClaraGenomicsAnalysis/blob/b5ea4f4d9d11ccf2bd8e8b4a571168ba6a996547/common/utils/include/claragenomics/utils/buffer.hpp#L204

I do agree this is a good practice as it future proofs for the case that `device_buffer::resize()` changes it's behavior and the price of calling `device_buffer::shrink_to_fit()` is negligible in this case

(so all good, just discussing)

Reply:
thanks for clarifying, I was not sure to add shrink_to_fit(), asked @akkamesh about it, he also suggested no harm to keep it.